### ISSUE TITLE
Error summary - accessibility and markup review

### ISSUE THREAD

**Issue by penx** [govuk-react/govuk-react#480]
There have been reports of the error summary component causing accessibility issues.

Error summary should:

- match govuk-frontend markup and CSS, or detail in this ticket if this is not feasible and why
- allow hierarchical headings, assuming govuk-frontend templates allow this, and following what govuk-frontend nunjucks templates do. If govuk-frontend doesn't do this, detail in this ticket
- allow anchors in error summary to be hrefs with in-page IDs that allow users to click through without JS, assuming this is already done by govuk-frontend

The above can be split in to separate tickets and/or PRs if we want to tackle individually.

Related feedback from FSA:

> 1. “There is a problem” on the error summary boxes is currently a fixed heading size. Heirarchical headings are good for accessibility, so control over this (while keeping the visible size the same) would be good.
> 2. We’ve used JS to make the error summary box links scroll to the correct errors. However, this actually doesn’t work with any screenreaders for some reason - real anchor hrefs would be preferable.

**Comment by abdulrah94** [comment]
another suggestion re accessibility in the ErrorSummary: it is not keyoard `focusable` nor is it keyboard `tabbable`

**Comment by penx** [comment]
As part of this, the Cypress tests for the example application should check the anchors link to the correct fields


https://github.com/govuk-react/govuk-react/pull/893/files#r662957639

**Comment by penx** [comment]
In addition, Cypress test examples should bring focus to the error summary component after an unsuccessful submission

See https://github.com/govuk-react/govuk-react/issues/886
